packages/runtime/src/tutorial-runner.ts

```diff
@@ -484,3 +521,19 @@ async function updateFiles(webcontainer: WebContainer, previousFiles: Files, new
 
   await webcontainer.mount(toFileTree(addedOrModified));
 }
+
+function removeLeadingSlash(filePath: string) {
+  if (filePath.startsWith('/')) {
+    return filePath.slice(1);
+  }
+
+  return filePath;
+}
+
+function parseJson(deserialized: undefined | string): any {
+  try {
+    return JSON.parse(deserialized || '{}');
```

Comment:
I find it surprising that parsing undefined returns an empty object.

In the code above where this function is used it would be more correct to leave the `package.json` unmodified if it isn't valid JSON. Otherwise this code will throw away everything else.

Maybe this auxiliary function is too generic. I think it's probably better to inline the logic.